lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -656,46 +664,65 @@ impl LayerModel {
         self.sublayers.borrow().all()
     }
 
-    /// Attach a `layer` as a sublayer.
+    /// Iterate over all sublayers, ordered according to the defined depth-order dependencies. The
+    /// layer sublayers list will be borrowed during the iteration.
+    ///
+    /// Please note that this function does not update the depth-ordering of the layers. Updates are
+    /// performed by calling the `update` method on [`Group`], which usually happens once per
+    /// animation frame.
+    pub fn for_each_sublayer(&self, mut f: impl FnMut(Layer)) {
```

Comment:
If this function borrows things, it should be probably named `with_borrowed_sublayers` to express that in the usage-place. Also, I do not see a reason for this function to exist TBH. Writing its body is as long as using this function and it clearly shows that we are bottowing sublayers.

Reply:
Replacing it that way would require `sublayers` field to be public. I think it's better to keep that function in place.